### PR TITLE
feat(dsg): add create main file event

### PR DESCRIPTION
**Pull request by Shurtu-gal**
Close: #7402

## PR Details

Adds createMainFile events.

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/data-service-generator/src/server/create-main/create-main-file.ts [hunk 1]
```diff
@@ -1,38 +1,50 @@
...
     __dirname,
     MAIN_TS_WITH_BIGINT_FILE_NAME
   );
-  const mainFileContent = await fs.readFile(mainFilePath, "utf-8");
-  const mainWithBigintFileContent = await fs.readFile(
-    mainWithBigintFilePath,
-    "utf-8"
-  );
+  const template = await readFile(mainFilePath);
+  const bigIntTemplate = await readFile(mainWithBigintFilePath);
```
Comment (overbit) [inline packages/data-service-generator/src/server/create-main/create-main-file.ts:20]: I'd recommend to move the logic that handle the template selection before so the plugin event would need to only take care of one template, making it simplier and more maintainable. @Shurtu-gal what do you think?

Comment (Shurtu-gal) [reply packages/data-service-generator/src/server/create-main/create-main-file.ts:20]: I could do that, however there is a difference between the two, which one should the template pass ?

diff: https://github.com/amplication/amplication/blob/2e4118c079dfc69b516598b843ec97fc7310ccfa/packages/data-service-generator/src/server/create-main/main.with-bigint.template.ts#L26-L31

Comment (overbit) [reply packages/data-service-generator/src/server/create-main/create-main-file.ts:20]: we can unify the two with the following patch 
[refactor-main.ts.patch](https://github.com/amplication/amplication/files/14981999/refactor-main.ts.patch)

Comment (Shurtu-gal) [reply packages/data-service-generator/src/server/create-main/create-main-file.ts:20]: So basically you want this to be applied in the AST tree, if I am able to understand it correctly.

Comment (Shurtu-gal) [reply packages/data-service-generator/src/server/create-main/create-main-file.ts]: Done @overbit.

**Comment by overbit** [comment]
well done @Shurtu-gal. 
Can you please update the snapshot too `npx nx update-snapshot data-service-generator`?
We also need to add the new event in out docs site https://github.com/amplication/docs

**Review by overbit — APPROVED** [review]

**Comment by Shurtu-gal** [comment]
> npx nx update-snapshot data-service-generator

@overbit done.

**Review by overbit — APPROVED** [review]

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**